addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -401,14 +401,10 @@ export function makeActionManager(env, router = _router) {
             if (target !== "inline" && !(target === "new" && action.views[0][1] === "form")) {
                 // FIXME: search view arch is already sent with load_action, so either remove it
                 // from there or load all fieldviews alongside the action for the sake of consistency
-                let searchViewId = false;
-                if (action.search_view_id) {
-                    searchViewId =
-                        action.search_view_id instanceof Object
-                            ? action.search_view_id.id
-                            : action.searchViewId[0];
-                }
-                action.views.push([searchViewId, "search"]);
+                const searchViewId = action.search_view_id?.[0]
+                    ? action.search_view_id[0]
+                    : action.search_view_id;
+                action.views.push([searchViewId || false, "search"]);
```

Comment:
I think we could just do that, instead of changing the behavior inside `_get_action_dict` and also here since we don't care of `display_name` for `search_view_id` field. 🤷🏼‍♂️ 

What do you think? 🙂